src/screenfull.js

```diff
@@ -93,13 +93,20 @@
 				// keyboard in fullscreen even though it doesn't.
 				// Browser sniffing, since the alternative with
 				// setTimeout is even worse.
+				let result;
 				if (/ Version\/5\.1(?:\.\d+)? Safari\//.test(navigator.userAgent)) {
-					elem[request]();
+					result = elem[request]();
 				} else {
-					elem[request](keyboardAllowed ? Element.ALLOW_KEYBOARD_INPUT : {});
+					result = elem[request](keyboardAllowed ? Element.ALLOW_KEYBOARD_INPUT : {});
 				}
-
-				this.on('change', onFullScreenEntered);
+				
+				if (!(result instanceof Promise)) {
+					result = Promise.resolve(result);
+				}
+				
+				return result.then(() => {
```

Comment:
You don't have to have the above conditional. Just resolve it no matter what.
```suggestion
				return Promise.resolve(result).then(() => {
```

Reply:
Returning inside a Promise constructor has no purpose.

Reply:
You cannot use arrow function.

Reply:
And you're not correctly handling the `result` promise rejection.